SparseGrids/tsgCoreOneDimensional.cpp

```diff
@@ -392,6 +393,72 @@ const char* OneDimensionalMeta::getHumanString(TypeOneDRule rule){
     }
 }
 
+// Generate the n roots of the n-th degree orthogonal polynomial.
+// ref_points and ref_points are used in the computation of the integrals that form the elements of the Jacobi matrix.
+double poly_eval(int nroots, std::vector<double> roots, double x) {
```

Comment:
Looking at this, is `nroots` different from `roots.size()`?

The vector keeps it's own size, which makes it easy to pass around.

Reply:
Then was an older version where ``roots`` was preallocated to have ``n`` elements. The new version has ``roots.size() == i+1`` for each level `i` now, so I have removed the need for ``nroots`` above.